Lib/tarfile.py

```diff
@@ -238,24 +238,32 @@ def copyfileobj(src, dst, length=None, exception=OSError, bufsize=None):
     """
     bufsize = bufsize or 16 * 1024
     if length == 0:
-        return
+        return 0
     if length is None:
         shutil.copyfileobj(src, dst, bufsize)
-        return
+        return None
 
+    copied_length = 0
     blocks, remainder = divmod(length, bufsize)
-    for b in range(blocks):
+    for _ in range(blocks):
         buf = src.read(bufsize)
         if len(buf) < bufsize:
-            raise exception("unexpected end of data")
+            if exception is not None:
+                raise exception("unexpected end of data")
+            print("warning: file changed as we read it")
```

Comment:
Do not use `print()`, use `warnings.warn` instead.

Reply:
Got it. I will use warnings.warn instead of print().